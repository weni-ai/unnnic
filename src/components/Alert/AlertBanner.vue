<template>
    <section :class="['banner-alert', {
        'banner-alert--banner-danger': type === 'danger',
        'banner-alert--banner-warning': type === 'warning',
      }]">
        <header v-show="text" class="banner-alert__container-text">
          <unnnic-icon class="banner-alert__textIcon" v-show="isShowTextIcon(type)" :icon="getIconType(type)" size="sm" scheme="neutral-white" />
          <p class="text" >{{ text }}</p>
          <a v-if="linkHref" class="banner-alert__link" :href="linkHref" :target="linkTarget">
            {{ linkText }}
          </a>
        </header>
  
        <div
          v-show="showCloseButton"
          class="banner-alert__close"
          @click="emitClose"
        >
          <unnnic-icon icon="close" size="sm" scheme="neutral-white" />
        </div>
      </section>
  </template>
  
  <script>
  import UnnnicIcon from '../Icon.vue';
  
  export default {
    components: {
      UnnnicIcon,
    },
    methods: {
      getIconType(type) {
        if (type === 'danger') return 'block';
        if (!type.trim()) return 'info'; //check if type is empty or whitespace
        return type;
      },
      isShowTextIcon(type) {
        return ['danger', 'warning', 'info', ''].includes(type)
      },
      emitClose() {
        this.onClose();

        this.$emit('close');
      }
    },
    props: {
      text: {
        type: String,
        default: null,
      }, 
      onClose: {
        type: Function,
        default: () => {}
      },
      showCloseButton: {
        type: Boolean,
        default: false
      },
      linkHref: {
        type: String,
        default: '',
      },
      linkTarget: {
        type: String,
        default: '_blank',
      },
      linkText: {
        type: String,
        default: 'Learn more',
      },
      type: {
        type: String,
        default: 'info',
      },
    }
  };
  </script>
  
  <style lang="scss" scoped>
  @import '../../assets/scss/unnnic.scss';
  
  .banner-alert {
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: space-between;
    margin-bottom: $unnnic-spacing-sm;
  
    color: $unnnic-color-neutral-white;
    font-family: $unnnic-font-family-secondary;
    font-size: $unnnic-font-size-body-gt;
    line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    font-weight: $unnnic-font-weight-regular;
    background-color: $unnnic-color-aux-blue-500;

    &__container-text {
      display: flex;
      justify-content: center;
      align-items: center;
      padding: $unnnic-spacing-xs $unnnic-spacing-sm;
      flex-grow: 1;

      .text {
        margin: 0 $unnnic-spacing-nano 0 0;
      }
    }

    &__textIcon {
      margin-right: $unnnic-spacing-xs;
    }
  
    &__link {
      color: inherit;
      display: block;
      font-weight: $unnnic-font-weight-bold;
      text-decoration: underline;
      text-underline-position: under;
    }
  
    &__close {
      display: flex;
      padding: 11px $unnnic-spacing-sm;
      cursor: pointer;
      user-select: none;
      border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-blue-300;
      &:hover {
          background-color: $unnnic-color-aux-blue-300;
        }
    }
  
    &--banner-warning {
      background-color: $unnnic-color-aux-orange-500;
  
      .banner-alert__close {
        border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-orange-300;
  
        &:hover {
          background-color: $unnnic-color-aux-orange-300;
        }
      }
    }
  
    &--banner-danger {
      background-color: $unnnic-color-aux-red-500;
  
      .banner-alert__close {
        border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-red-300;
  
        &:hover {
          background-color: $unnnic-color-aux-red-300;
        }
      }
    }
  }
  </style>
  