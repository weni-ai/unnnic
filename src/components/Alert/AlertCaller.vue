<template>
    <button @click="callAlert"> Launch alert </button>
</template>

<script>
import alert from '../../utils/call';

export default {
  props: {
    title: {
      type: String,
      default: null,
    },
    text: {
      type: String,
      default: null,
    },
    icon: {
      type: String,
      default: null,
    },
    enabled: {
      type: Boolean,
      default: true,
    },
    scheme: {
      type: String,
      default: null,
    },
    seconds: {
      type: Number,
      default: 3,
    },
    closeText: {
      type: String,
      default: null,
    },
    position: {
      type: String,
      default: 'top-right',
    },
  },
  methods: {
    callAlert() {
      alert.callAlert({ props: this.$props, seconds: this.seconds });
    },
  },
};
</script>
