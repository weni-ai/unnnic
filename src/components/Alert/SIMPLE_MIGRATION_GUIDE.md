# üöÄ Guia de Migra√ß√£o: Alert ‚Üí Toast

## üì¶ Importa√ß√£o

**ANTES:**

```javascript
import Unnnic from '@weni/unnnic-system';

Unnnic.unnnicCallAlert(...);
```

**DEPOIS:**

```javascript
import { toast } from "@weni/unnnic-system";
```

---

## üìã Mapeamento de Propriedades

| Alert               | Toast               | Altera√ß√£o                       |
| ------------------- | ------------------- | ------------------------------- |
| `text`              | `title`             | Propriedade renomeada           |
| `seconds`           | `timeout`           | Agora em **milissegundos (ms)** |
| `type: 'success'`   | `toast.success()`   | Substitu√≠do por m√©todo direto   |
| `type: 'error'`     | `toast.error()`     | Substitu√≠do por m√©todo direto   |
| `type: 'attention'` | `toast.attention()` | Substitu√≠do por m√©todo direto   |
| `type: 'default'`   | `toast.info()`      | Substitu√≠do por m√©todo direto   |

---

## üîÑ Exemplos de Uso

### Exemplo 1 ‚Äì Uso com Fun√ß√£o

**ANTES:**

```javascript
// callAlert
alert.callAlert({
  props: {
    text: "Dados salvos com sucesso!",
    type: "success",
  },
  seconds: 3,
});
```

**DEPOIS:**

```javascript
// toast
toast.success("Dados salvos com sucesso!");
```

---

### Exemplo 2 ‚Äì Uso com Componente

**ANTES:**

```vue
<!-- UnnnicAlert -->
<UnnnicAlert
  text="Dados salvos com sucesso!"
  type="success"
  @close="handleClose"
/>
```

**DEPOIS:**

```vue
<!-- UnnnicToast -->
<UnnnicToast
  title="Dados salvos com sucesso!"
  type="success"
  @close="handleClose"
/>
```

---

## üìö Casos Comuns

### ‚úÖ Sucesso

**ANTES:**

```javascript
alert.callAlert({
  props: {
    text: "Dados salvos com sucesso!",
    type: "success",
  },
  seconds: 5,
});
```

**DEPOIS:**

```javascript
toast.success("Dados salvos com sucesso!");
```

---

### ‚ùå Erro

**ANTES:**

```javascript
alert.callAlert({
  props: {
    text: "Erro ao salvar dados",
    type: "error",
  },
  seconds: 5,
});
```

**DEPOIS:**

```javascript
toast.error("Erro ao salvar dados");
```

---

### ‚è± Timeout Personalizado

**ANTES:**

```javascript
alert.callAlert({
  props: {
    text: "Mensagem importante",
    type: "success",
  },
  seconds: 10, // 10 segundos
});
```

**DEPOIS:**

```javascript
toast.success("Mensagem importante", "", {
  timeout: 10000, // 10 segundos em ms
});
```
