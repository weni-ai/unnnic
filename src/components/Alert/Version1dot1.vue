<template>
  <div ref="alertContainer" class="alert-container">
    <div :class="['alert', {
      'alert--scheme-aux-green': scheme === 'feedback-green' || type === 'success',
      'alert--scheme-aux-red': scheme === 'feedback-red' || type === 'error',
    }]">
      <div ref="progress" class="alert__progress"></div>

      <div v-show="text" class="alert__text">
        {{ text }}
      </div>

      <a v-if="linkHref" class="alert__link" :href="linkHref" :target="linkTarget">
        {{ linkText }}
      </a>

      <div
        class="alert__close"
        @click="emitClose"
      >
        <unnnic-icon icon="close-1" size="sm" scheme="neutral-white" />
      </div>
    </div>
  </div>
</template>

<script>
import UnnnicIcon from '../Icon.vue';

export default {
  components: {
    UnnnicIcon,
  },
  props: {
    version: {
      type: String,
      default: '1.0',
    },
    text: {
      type: String,
      default: null,
    },
    scheme: {
      type: String,
      default: null,
    },
    onClose: {
      type: Function,
      default: () => {},
    },
    linkHref: {
      type: String,
      default: '',
    },
    linkTarget: {
      type: String,
      default: '_blank',
    },
    linkText: {
      type: String,
      default: 'Learn more',
    },
    type: {
      type: String,
      default: 'default',
    },
  },

  mounted() {
    this.$refs.progress.addEventListener('animationend', () => {
      this.$refs.alertContainer.classList.add('slide-down');
    });

    this.$refs.alertContainer.addEventListener('animationend', (event) => {
      if (event.animationName.startsWith('slideDown')) {
        this.emitClose();
      }
    });
  },

  methods: {
    emitClose() {
      this.onClose();

      this.$emit('close');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../../assets/scss/unnnic.scss';

.alert-container {
  position: fixed;
  bottom: 0;
  left: 0;
  width: 100%;
  text-align: center;
  animation: slideUp 200ms ease;

  @keyframes slideUp {
    from {
      bottom: -3.375 * $unnnic-font-size;
    }

    to {
      bottom: 0;
    }
  }

  &.slide-down {
    animation-name: slideDown;
  }

  @keyframes slideDown {
    from {
      bottom: 0;
    }

    to {
      bottom: -3.375 * $unnnic-font-size;
    }
  }
}

.alert {
  position: relative;
  border-radius: $unnnic-border-radius-sm;

  display: inline-flex;
  align-items: center;
  margin-bottom: $unnnic-spacing-sm;

  color: $unnnic-color-neutral-white;
  font-family: $unnnic-font-family-secondary;
  font-size: $unnnic-font-size-body-gt;
  line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
  font-weight: $unnnic-font-weight-regular;

  background-color: $unnnic-color-neutral-dark;

  z-index: 9999;

  overflow: hidden;

  @keyframes progress {
    from {
      width: 0;
    }

    to {
      width: 100%;
    }
  }

  &__progress {
    position: absolute;
    height: 100%;
    background-color: $unnnic-color-neutral-darkest;
    border-radius: $unnnic-border-radius-sm;
    left: 0;
    top: 0;
    pointer-events: none;
    z-index: -1;
    animation-name: progress;
    animation-duration: 5000ms;
    animation-timing-function: linear;
  }

  &:hover {
    .alert__progress {
      animation-play-state: paused;
    }
  }

  &__text {
    padding: $unnnic-spacing-xs $unnnic-spacing-sm;
  }

  &__link, &__close {
    border-left: $unnnic-border-width-thinner solid $unnnic-color-neutral-cloudy;

    &:hover {
      background-color: $unnnic-color-neutral-cloudy;
    }
  }

  &__link {
    padding: $unnnic-spacing-xs $unnnic-spacing-sm;
    text-decoration: none;
    color: inherit;
    display: block;
    font-weight: $unnnic-font-weight-bold;
  }

  &__close {
    padding: $unnnic-spacing-xs $unnnic-spacing-sm;
    cursor: pointer;
    user-select: none;
  }

  &--scheme-aux-green {
    background-color: $unnnic-color-aux-green-500;

    .alert__progress {
      background-color: $unnnic-color-aux-green-700;
    }

    .alert__link, .alert__close {
      border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-green-300;

      &:hover {
        background-color: $unnnic-color-aux-green-300;
      }
    }
  }

  &--scheme-aux-red {
    background-color: $unnnic-color-aux-red-500;

    .alert__progress {
      background-color: $unnnic-color-aux-red-700;
    }

    .alert__link, .alert__close {
      border-left: $unnnic-border-width-thinner solid $unnnic-color-aux-red-300;

      &:hover {
        background-color: $unnnic-color-aux-red-300;
      }
    }
  }
}
</style>
