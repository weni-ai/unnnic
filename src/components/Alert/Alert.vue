<template>
  <div v-if="version === '1.0'" :class="['unnnic-alert', `unnnic-alert-position--${position}`]">
    <unnnic-icon-svg
      :icon="icon"
      :scheme="scheme"
      size="sm"
    />
    <div class="unnnic-alert__content">
      <div class="unnnic-alert__title">
        {{ title.toUpperCase() }}
      </div>
      <div class="unnnic-alert__text">
        {{ text }}
      </div>
    </div>
    <div
      v-if="closeText"
      class="unnnic-alert__close-text unnnic--clickable"
      @click="onClose()">
      {{ closeText.toUpperCase() }}
    </div>
    <unnnic-icon-svg
      v-else
      clickable
      icon="close-1"
      scheme="brand-sec"
      size="xs"
      @click="onClose"
    />
  </div>

  <component
    v-else :is="'version' + version.replace(/\./g, 'dot')"
    v-bind="{ ...$props }"
    @close="$emit('close')"
  ></component>
</template>

<script>
import unnnicIconSvg from '../Icon.vue';
import Version1dot1 from './Version1dot1.vue';

export default {
  name: 'unnnicAlert',
  components: {
    unnnicIconSvg,
    Version1dot1,
  },
  props: {
    version: {
      type: String,
      default: '1.1',
    },

    title: {
      type: String,
      default: null,
    },
    text: {
      type: String,
      default: null,
    },
    scheme: {
      type: String,
      default: null,
    },
    icon: {
      type: String,
      default: null,
    },
    onClose: {
      type: Function,
      default: () => {},
    },
    closeText: {
      type: String,
    },
    position: {
      type: String,
      default: 'top-right',
    },
    linkHref: {
      type: String,
    },
    linkTarget: {
      type: String,
    },
    linkText: {
      type: String,
    },
    type: {
      type: String,
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../../assets/scss/unnnic.scss';

  .unnnic-alert {
    position: fixed;
    padding: $unnnic-inset-xs;
    min-width: 12.5 * $unnnic-font-size;

    display: inline-flex;
    align-items: center;

    font-family: $unnnic-font-family-secondary;
    border-radius: $unnnic-border-radius-sm;

    background-color: $unnnic-color-background-snow;
    box-shadow: $unnnic-shadow-level-near;

    z-index: 9999;

    &-position {
      &--top-right {
        top: 1 * $unnnic-font-size;
        right: 1 * $unnnic-font-size;
      }
      &--top-left {
        top: 1 * $unnnic-font-size;
        left: 1 * $unnnic-font-size;
      }
      &--bottom-right {
        bottom: 1 * $unnnic-font-size;
        right: 1 * $unnnic-font-size;
      }
      &--bottom-left {
        bottom: 1 * $unnnic-font-size;
        left: 1 * $unnnic-font-size;
      }
    }

    &__content {
      flex: 1;
      margin: 0 $unnnic-inline-xs;
    }

    &__title {
      font-size: $unnnic-font-size-body-sm ;
      font-weight: $unnnic-font-weight-bold;
      color: $unnnic-color-neutral-darkest;
    }

    &__text {
      font-size: $unnnic-font-size-body-md;
      font-weight: $unnnic-font-weight-regular;
      color: $unnnic-color-neutral-dark;
    }

    &__close {
      &-text {
        color: $unnnic-color-brand-sec;
        font-size: $unnnic-font-size-body-md;
      }
    }
  }
</style>
