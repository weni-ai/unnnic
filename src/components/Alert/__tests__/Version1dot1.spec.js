import { mount } from '@vue/test-utils';
import { describe, expect, test, vi, beforeEach, afterEach } from 'vitest';
import Alert from '@/components/Alert/Version1dot1.vue';
import UnnnicIcon from '@/components/Icon.vue';

vi.mock('@/components/Icon.vue');

describe('Alert.vue', () => {
  let wrapper;

  const defaultProps = {
    version: '1.0',
    text: 'Test Alert',
    scheme: 'feedback-green',
    onClose: vi.fn(),
    linkHref: 'https://example.com',
    linkTarget: '_blank',
    linkText: 'Learn more',
    type: 'success',
  };

  beforeEach(() => {
    wrapper = mount(Alert, {
      props: defaultProps,
      global: {
        stubs: {
          UnnnicIcon: true,
        },
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
    vi.clearAllMocks();
  });

  test('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find('.alert').exists()).toBe(true);
    expect(wrapper.find('.alert__text').text()).toBe('Test Alert');
  });

  test('applies correct classes based on scheme and type props', () => {
    expect(wrapper.find('.alert--scheme-aux-green').exists()).toBe(true);

    wrapper = mount(Alert, {
      props: { ...defaultProps, scheme: 'feedback-red', type: 'error' },
      global: { stubs: { UnnnicIcon: true } },
    });
    expect(wrapper.find('.alert--scheme-aux-red').exists()).toBe(true);
  });

  test('renders link when linkHref is provided', () => {
    const link = wrapper.find('.alert__link');
    expect(link.exists()).toBe(true);
    expect(link.attributes('href')).toBe('https://example.com');
    expect(link.attributes('target')).toBe('_blank');
    expect(link.text()).toBe('Learn more');
  });

  test('does not render link when linkHref is not provided', async () => {
    await wrapper.setProps({ linkHref: '' });
    expect(wrapper.find('.alert__link').exists()).toBe(false);
  });

  test('emits close event and calls onClose prop when close button is clicked', async () => {
    await wrapper.find('.alert__close').trigger('click');
    expect(wrapper.emitted('close')).toBeTruthy();
    expect(defaultProps.onClose).toHaveBeenCalled();
  });

  test('adds slide-down class on progress animation end', async () => {
    const progressElement = wrapper.find('.alert__progress');
    await progressElement.trigger('animationend');
    expect(wrapper.find('.alert-container').classes()).toContain('slide-down');
  });

  test('emits close event on slide-down animation end', async () => {
    const alertContainer = wrapper.find('.alert-container');
    await alertContainer.trigger('animationend', {
      animationName: 'slideDown',
    });
    expect(wrapper.emitted('close')).toBeTruthy();
  });

  test('does not emit close event on non-slideDown animation end', async () => {
    const alertContainer = wrapper.find('.alert-container');
    await alertContainer.trigger('animationend', {
      animationName: 'someOtherAnimation',
    });
    expect(wrapper.emitted('close')).toBeFalsy();
  });

  test.each([
    ['version', '1.1'],
    ['text', 'New Alert Text'],
    ['scheme', 'feedback-red'],
    ['linkHref', 'https://newexample.com'],
    ['linkTarget', '_self'],
    ['linkText', 'Click here'],
    ['type', 'error'],
  ])('updates %s prop correctly', async (propName, newValue) => {
    await wrapper.setProps({ [propName]: newValue });
    expect(wrapper.props(propName)).toBe(newValue);
  });

  test('renders UnnnicIcon component', () => {
    expect(wrapper.findComponent(UnnnicIcon).exists()).toBe(true);
    expect(wrapper.findComponent(UnnnicIcon).props()).toEqual({
      clickable: false,
      filled: false,
      icon: 'close-1',
      size: 'sm',
      scheme: 'neutral-white',
    });
  });

  test('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
