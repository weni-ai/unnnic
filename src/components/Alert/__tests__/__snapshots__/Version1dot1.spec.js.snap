// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert.vue > matches snapshot 1`] = `
"<div data-v-fb94f284="" class="alert-container" version="1.0" linkhref="https://example.com" linktarget="_blank" linktext="Learn more">
  <div data-v-fb94f284="" class="alert alert--scheme-aux-green">
    <div data-v-fb94f284="" class="alert__progress"></div>
    <div data-v-fb94f284="" class="alert__text">Test Alert</div>
    <div data-v-fb94f284="" class="alert__close">
      <unnnic-icon-stub data-v-fb94f284="" filled="false" next="false" icon="close-1" clickable="false" size="sm" scheme="neutral-white"></unnnic-icon-stub>
    </div>
  </div>
</div>"
`;
